engine/src/main/java/org/terasology/engine/subsystem/openvr/OpenVRControllers.java

```diff
@@ -51,90 +59,83 @@
         queuedActions.clear();
         return result;
     }
 
-    private void handleController0(VRControllerState_t stateBefore, VRControllerState_t stateAfter) {
-        if (OpenVRUtil.switchedUp(ControllerListener.BUTTON_TRIGGER, stateBefore.ulButtonPressed, stateAfter.ulButtonPressed)) {
-            queuedActions.add(new ControllerAction(InputType.CONTROLLER_BUTTON.getInput(ControllerId.ZERO),
-                    "OpenVR", ButtonState.UP, 1.0f));
-        } else if (OpenVRUtil.switchedDown(ControllerListener.BUTTON_TRIGGER, stateBefore.ulButtonPressed, stateAfter.ulButtonPressed)) {
-            queuedActions.add(new ControllerAction(InputType.CONTROLLER_BUTTON.getInput(ControllerId.ZERO),
-                    "OpenVR", ButtonState.DOWN, 1.0f));
-        } else if (OpenVRUtil.switchedUp(ControllerListener.BUTTON_GRIP, stateBefore.ulButtonPressed, stateAfter.ulButtonPressed)) {
-            queuedActions.add(new ControllerAction(InputType.CONTROLLER_BUTTON.getInput(ControllerId.ONE),
-                    "OpenVR", ButtonState.UP, 1.0f));
-        } else if (OpenVRUtil.switchedDown(ControllerListener.BUTTON_GRIP, stateBefore.ulButtonPressed, stateAfter.ulButtonPressed)) {
-            queuedActions.add(new ControllerAction(InputType.CONTROLLER_BUTTON.getInput(ControllerId.ONE),
-                    "OpenVR", ButtonState.DOWN, 1.0f));
-        } else if (OpenVRUtil.switchedUp(ControllerListener.BUTTON_APP_MENU, stateBefore.ulButtonPressed, stateAfter.ulButtonPressed)) {
-            queuedActions.add(new ControllerAction(InputType.CONTROLLER_BUTTON.getInput(ControllerId.TWO),
-                    "OpenVR", ButtonState.UP, 1.0f));
-        } else if (OpenVRUtil.switchedDown(ControllerListener.BUTTON_APP_MENU, stateBefore.ulButtonPressed, stateAfter.ulButtonPressed)) {
-            queuedActions.add(new ControllerAction(InputType.CONTROLLER_BUTTON.getInput(ControllerId.TWO),
-                    "OpenVR", ButtonState.DOWN, 1.0f));
-        } else if (OpenVRUtil.switchedDown(ControllerListener.BUTTON_TOUCHPAD, stateBefore.ulButtonPressed, stateAfter.ulButtonPressed)) {
-            queuedActions.add(new ControllerAction(InputType.CONTROLLER_AXIS.getInput(ControllerId.X_AXIS),
-                    "OpenVR", ButtonState.DOWN, -stateAfter.rAxis[0].x));
-            queuedActions.add(new ControllerAction(InputType.CONTROLLER_AXIS.getInput(ControllerId.Y_AXIS),
-                    "OpenVR", ButtonState.DOWN, stateAfter.rAxis[0].y));
-        } else if (OpenVRUtil.switchedUp(ControllerListener.BUTTON_TOUCHPAD, stateBefore.ulButtonPressed, stateAfter.ulButtonPressed)) {
-            queuedActions.add(new ControllerAction(InputType.CONTROLLER_AXIS.getInput(ControllerId.X_AXIS),
-                    "OpenVR", ButtonState.UP, 0));
-            queuedActions.add(new ControllerAction(InputType.CONTROLLER_AXIS.getInput(ControllerId.Y_AXIS),
-                    "OpenVR", ButtonState.UP, 0));
+    private boolean switchedUp(long buttonIndex) {
```

Comment:
Could this be `buttonUp`? When I read this I just keep wondering _what_ is switching up. And could buttonIndex be `buttonId`?